### ISSUE TITLE
Which keypoints are actually detected?

### ISSUE THREAD

**Issue by rohitrango** [facebookresearch/Detectron#737]
Hi,

I have a small question. During evaluation, the detectron outputs all the keypoints' location, and some logit and probability associated with it. There is no binary tag regarding if the key point is detected or not. I want to test a custom algorithm that uses the information of the joints that are not detected. How can I know whether a keypoint is detected or not (and hence be able to assign a hard binary number to its existence)? 

Thanks in advance.

**Comment by ir413** [comment]
Hi @rohitrango, for COCO style evaluation all keypoints for an object are passed together with an object-level confidence score. Predicted keypoints that are not labeled in the ground-truth do not affect the COCO OKS metric. Please see [this page](http://cocodataset.org/#keypoints-eval) for more info on COCO keypoint evaluation.

Since in addition to keypoint locations detectron outputs per keypoint scores, you can try to estimate if a keypoint is detected or not by thresholding based on the scores (e.g. similarly to what is done when visualizing the predicted keypoints [here](https://github.com/facebookresearch/Detectron/blob/master/detectron/utils/vis.py#L343))

**Comment by rohitrango** [comment]
I see. However, there's a threshold on the logit value, which doesn't make a lot of sense, since the softmax function can output the same probability vectors for translated logits, and the threshold might be different for different trained models. Is there something I'm missing here or is it the way how it works?

 I'm clear about the OKs metric though.
